site/_collections/algolia.js

```diff
@@ -62,7 +62,7 @@ const algoliaCollection = collections => {
       data.disable_algolia ||
       data.noindex ||
       data.draft ||
-      data.permalink === false
+      (data.permalink === false && data.canLinkToList !== true)
```

Comment:
LEt's use some more generic name for this property. We have VirtualCollectionItem type to describe item that do now have a corresponsing .md file and therefore permalink. Maybe .isVirtualItem = true to make this indexing change more generic?